File: octopoints-kotlin/app/src/main/java/dev/tpavani/octopoints/db/dao/MatchDao.kt
package dev.tpavani.octopoints.db.dao

import androidx.room.Dao
import androidx.room.Query
import dev.tpavani.octopoints.db.entity.MatchEntity

@Dao
interface MatchDao : IDao<MatchEntity> {
    @Query("SELECT * FROM matches ORDER BY name")
    suspend fun getMatches(): List<MatchEntity>
}

File: octopoints_flutter/lib/db/dao/MatchDao.dart
import 'package:floor_annotation/floor_annotation.dart';
import 'package:octopoints_flutter/db/dao/IDao.dart';
import 'package:octopoints_flutter/db/entity/MatchEntity.dart';

@dao
abstract class MatchDao extends IDao<MatchEntity> {
  @Query('SELECT * FROM matches ORDER BY name')
  Future<List<MatchEntity>> getMatches();
}
